classes/CartRule.php

```diff
@@ -96,6 +97,7 @@ class CartRuleCore extends ObjectModel
         'multilang' => true,
         'fields' => [
             'id_customer' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId'],
+            'id_order_invoice' => ['type' => self::TYPE_INT, 'validate' => 'isUnsignedId'],
```

Comment:
This is frightening me. Introducing a new field and SQL column into the amazingly complex data model of Cart Rules. We know there are some complex SQL queries doing massive checks (see https://github.com/PrestaShop/PrestaShop/issues/19935) impacted by number of rows.

This change will not be beta-tested (unless there is a beta 3). I think it's too risky to be releases with PS 1.7.7.0 .

Reply:
I get why you are not reassured ^^ However the risk here is really contained because this extra field is only used internally for cart rules that are created programmatically by the core handlers. And it is only used in the multi-invoice "mode", so 95% of the time it won't even be used by merchants.

I didn't change any of the existing queries so it can't impact them, this field is only used to perform a filtering in the PHP side (which prevented from modifying the queries).

However, currently we have a regression because it's not possible any more to create discounts on a specific invoice (see this issue https://github.com/PrestaShop/PrestaShop/issues/20778 )